src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -267,6 +267,27 @@ fn check_contract_call(checker: &mut TypeChecker, args: &[SymbolicExpression], c
     Ok(expected_sig.returns)
 }
 
+fn check_contract_of(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
+    check_argument_count(1, args)?;
+
+    let trait_instance = match &args[0].expr {
+        SymbolicExpressionType::Atom(trait_instance) => trait_instance,
+        _ => return Err(CheckError::new(CheckErrors::ContractOfExpectsTrait))
+    };
+
+    let trait_id = match context.lookup_trait_reference_type(trait_instance) {
```

Comment:
Maybe more of a question for @kantai or @lgalabru, but I want to confirm that `lookup_trait_reference_type` is both I/O-free and O(1)?  If not, then the `runtime_cost!` macro belongs above this line.

Reply:
or `lookup_variable` would also need to be changed

Reply:
Yes, that's correct, `lookup_trait_reference_type` is both I/O free and `O(1)` -- trait references may only be created in the top-level context (so, an `O(1)` lookup), and the trait reference isn't cloned, so it doesn't depend on the trait size either.

Reply:
Thanks!